cli/compose/loader/loader_test.go

```diff
@@ -1395,3 +1395,60 @@ networks:
 	}
 	assert.DeepEqual(t, config, expected, cmpopts.EquateEmpty())
 }
+
+func TestLoadInit(t *testing.T) {
+	booleanFalse := false
+	booleanTrue := true
+	config, err := loadYAML(`
+version: '3.7'
+services:
+  foo:
+    image: alpine`)
+	assert.NilError(t, err)
+	assert.Check(t, is.DeepEqual(config, &types.Config{
+		Filename: "filename.yml",
+		Version:  "3.7",
+		Services: types.Services{
+			{
+				Name:  "foo",
+				Image: "alpine",
+			},
+		},
+	}, cmpopts.EquateEmpty()))
+	config, err = loadYAML(`
+version: '3.7'
+services:
+  foo:
+    image: alpine
+    init: true`)
+	assert.NilError(t, err)
+	assert.Check(t, is.DeepEqual(config, &types.Config{
```

Comment:
I think you can just check the Init value, checking the equality on all the Config is a little bit extreme here:
```go
assert.Equal(t, isconfig.Services[0].Init, &booleanTrue)
```
You may argue that then you have to check Services length before, but I think we can skip it 😇

Reply:
Then you can refactor the test with these 3 cases using Table Tests.